tests/test_templates.py

```diff
@@ -12,106 +12,249 @@
+        templ1(*inp1_, wires=range(2))
+        templ2(*inp2_, wires=range(2))
+        qml.PauliX(wires=1)
+        return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+    return circuit
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            StronglyEntanglingLayer(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
 
-        circuit(weights=p)
+def qnode_cv_args(dev, intrfc, templ1, templ2, n):
+    """QNode for CV integration circuit using positional arguments"""
+    @qml.qnode(dev, interface=intrfc)
+    def circuit(*inp_):
+        # Split inputs again
+        inp1_ = inp_[:n]
+        inp2_ = inp_[n:]
 
-    @pytest.mark.parametrize('parfun', [random_layers_uniform, random_layers_normal])
-    def test_integration_random_layers(self, parfun, qubit_device, n_subsystems, n_layers):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.RandomLayers()."""
+        qml.Displacement(1., 1., wires=0)
+        templ1(*inp1_, wires=range(2))
+        templ2(*inp2_, wires=range(2))
+        qml.Displacement(1., 1., wires=1)
+        return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
+    return circuit
 
-        p = parfun(n_layers=n_layers, n_wires=n_subsystems)
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            RandomLayers(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+def qnode_cv_kwargs(dev, intrfc, templ1, templ2, n):
+    """QNode for CV integration circuit using keyword arguments"""
+    @qml.qnode(dev, interface=intrfc)
+    def circuit(**inp_):
+        # Split inputs again
+        ks = [int(k) for k in inp_.keys()]
+        vs = inp_.values()
+        inp_ = [x for _, x in sorted(zip(ks, vs))]
+        inp1_ = inp_[:n]
+        inp2_ = inp_[n:]
+
+        qml.Displacement(1., 1., wires=0)
+        templ1(*inp1_, wires=range(2))
+        templ2(*inp2_, wires=range(2))
+        qml.Displacement(1., 1., wires=1)
+        return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
+    return circuit
+
+######################
+
+
+class TestIntegrationCircuit:
+    """Tests the integration of templates into circuits using the NumPy interface. """
+
+    @pytest.mark.parametrize("template1, inpts1", qubit_const)
+    @pytest.mark.parametrize("template2, inpts2", qubit_const)
+    @pytest.mark.parametrize("intrfc, to_var", interfaces)
+    def test_integration_qubit_args(self, template1, inpts1, template2, inpts2,
+                                    intrfc, to_var):
+        """Checks integration of qubit templates using positional arguments."""
+        inpts = inpts1 + inpts2  # Combine inputs to allow passing with *
+        inpts = [to_var(i) for i in inpts]
+        dev = qml.device('default.qubit', wires=2)
+        circuit = qnode_qubit_args(dev, intrfc, template1, template2, len(inpts1))
+        circuit(*inpts)
```

Comment:
This test isn't asserting anything

Reply:
Re: `circuit(inpts)`, I think worth testing, as you can probably copy and paste this test and make slight adjustments

Reply:
Added comment